tests/write-json.test.js

```diff
@@ -0,0 +1,91 @@
+const { promises: fs } = require('fs');
+const { convertToJson } = require('../scripts/utils');
+const writeJSON = require("../scripts/utils/write-json");
+const {yamlString, jsonObject} = require("./fixtures/utilsData");
+
+jest.mock('fs', () => ({
+  promises: {
+    readFile: jest.fn(),
+    writeFile: jest.fn(),
+  },
+}));
+
+jest.mock('../scripts/utils', () => ({
+  convertToJson: jest.fn(),
+}));
+
+describe('writeJSON', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('should read a file, convert it to JSON, and write the JSON to another file', async () => {
+    const readPath = 'config/testInput.yaml';
+    const writePath = 'config/testOutput.json';
+
+    fs.readFile.mockResolvedValue(yamlString);
+    convertToJson.mockReturnValue(jsonObject);
+
+    await writeJSON(readPath, writePath);
+
+    expect(fs.readFile).toHaveBeenCalledWith(readPath, 'utf-8');
+    expect(convertToJson).toHaveBeenCalledWith(yamlString);
+    expect(fs.writeFile).toHaveBeenCalledWith(writePath, JSON.stringify(jsonObject));
+  });
+
+  test('should log an error and throw if reading the file fails', async () => {
+    const readPath = 'config/testInput.yaml';
+    const writePath = 'config/testOutput.json';
+    const error = new Error('File read error');
+
+    fs.readFile.mockRejectedValue(error);
+    const consoleErrorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
+
+    await expect(writeJSON(readPath, writePath)).rejects.toThrow(error);
+
+    expect(consoleErrorSpy).toHaveBeenCalledWith(
+      `Error reading file from path ${readPath}:`,
+      error
+    );
+    consoleErrorSpy.mockRestore();
```

Comment:
Do we need this as we already cleared the mocks in `beforeEach` callback?

Reply:
yes we already have it in beforeEach and it works well even if i remove it but clearAllMocks does not restore it to its original implementations so i added it